apps/storage/api/attachments/internalPresign.get.ts

```diff
@@ -0,0 +1,42 @@
+import { PutObjectCommand } from '@aws-sdk/client-s3';
+import { getSignedUrl } from '@aws-sdk/s3-request-presigner';
+import { db } from '@uninbox/database';
+import { and, eq } from '@uninbox/database/orm';
+import { orgMembers, orgs } from '@uninbox/database/schema';
+import { nanoId } from '@uninbox/utils';
+import { z } from 'zod';
+
+const bodySchema = z.object({
+  orgPublicId: z.string(),
+  filename: z.string()
+});
+/**
+ * Returns a presigned URL for uploading an attachment to S3
+ */
+export default eventHandler({
+  onRequest: [authorizedService],
+  async handler(event) {
+    const inputValidation = await readValidatedBody(event, (body) =>
+      bodySchema.safeParse(body)
+    );
+    if (!inputValidation.success) {
+      setResponseStatus(event, 400);
+      return send(event, 'Invalid input');
+    }
```

Comment:
Can you check if we can just throw like this https://unjs.io/blog/2023-08-15-h3-towards-the-edge-of-the-web#runtime-type-safe-request-utils

Reply:
got some type errors when trying to do that, couldnt access the error